File: src/Java/ImageData/ImageNormalize.java
import vtk.vtkNativeLibrary;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;
import vtk.vtkImageSinusoidSource;
import vtk.vtkNamedColors;
import vtk.vtkImageNormalize;
import vtk.vtkImageCast;
import vtk.vtkImageActor;


public class ImageNormalize 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {
  
    vtkNamedColors colors = new vtkNamedColors();
    
    double left_bkg[] = new double[]{0.6, 0.5, 0.4, 1.0};
    double right_bkg[] = new double[]{0.4, 0.5, 0.6, 1.0};
    
    colors.SetColor("bkgColor", left_bkg);
    colors.SetColor("boxColor", right_bkg);
    
    //  Create an image
    vtkImageSinusoidSource source = new vtkImageSinusoidSource();
    source.Update();

    vtkImageNormalize normalizeFilter = new vtkImageNormalize();

    normalizeFilter.SetInputConnection(source.GetOutputPort());
    normalizeFilter.Update();
    
    vtkImageCast inputCastFilter = new vtkImageCast();
    inputCastFilter.SetInputConnection(source.GetOutputPort());
    inputCastFilter.SetOutputScalarTypeToUnsignedChar();
    inputCastFilter.Update();

    vtkImageCast normalizeCastFilter = new vtkImageCast();
    normalizeCastFilter.SetInputConnection(normalizeFilter.GetOutputPort());
    normalizeCastFilter.SetOutputScalarTypeToUnsignedChar();
    normalizeCastFilter.Update();
    
    //  Create actors
    vtkImageActor inputActor = new vtkImageActor();
    inputActor.GetMapper().SetInputConnection(inputCastFilter.GetOutputPort());

    vtkImageActor normalizedActor = new vtkImageActor();
    normalizedActor.GetMapper().SetInputConnection(normalizeCastFilter.GetOutputPort());
    
    //  Create the render window and interactor.
    
    vtkRenderWindow renWin = new vtkRenderWindow();
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    iren.SetRenderWindow(renWin);
    
    //  Define viewport ranges
    // (xmin, ymin, xmax, ymax)
    double leftViewport[] = new double[] {0.0, 0.0, 0.5, 1.0};
    double rightViewport[] = new double[] {0.5, 0.0, 1.0, 1.0};
    
    
    //  Setup both renderers
    vtkRenderer leftRenderer = new vtkRenderer();
    renWin.AddRenderer(leftRenderer);
    leftRenderer.SetViewport(leftViewport);
    leftRenderer.SetBackground(left_bkg);

    vtkRenderer rightRenderer = new vtkRenderer();
    renWin.AddRenderer(rightRenderer);
    rightRenderer.SetViewport(rightViewport);
    rightRenderer.SetBackground(right_bkg);
    
    leftRenderer.AddActor(inputActor);
    rightRenderer.AddActor(normalizedActor);

    leftRenderer.ResetCamera();
    rightRenderer.ResetCamera();
    
    renWin.SetSize(600,300);
    renWin.Render();
   
    iren.Initialize();
    iren.Start();
  }
}

File: src/Python/ImageData/ImageNormalize.py
#!/usr/bin/env python

import vtk

def main():
    colors = vtk.vtkNamedColors()
    colors.SetColor("right_bkg", [0.4, 0.5, 0.6, 1.0])
    colors.SetColor("left_bkg", [0.6, 0.5, 0.4, 1.0])
    
	# Create an image
    source = vtk.vtkImageSinusoidSource()
    source.Update()
    
    normalizeFilter = vtk.vtkImageNormalize()
    
    normalizeFilter.SetInputConnection(source.GetOutputPort())
    normalizeFilter.Update()
    
    inputCastFilter = vtk.vtkImageCast()
    inputCastFilter.SetInputConnection(source.GetOutputPort())
    inputCastFilter.SetOutputScalarTypeToUnsignedChar()
    inputCastFilter.Update()
    
    normalizeCastFilter = vtk.vtkImageCast()
    normalizeCastFilter.SetInputConnection(normalizeFilter.GetOutputPort())
    normalizeCastFilter.SetOutputScalarTypeToUnsignedChar()
    normalizeCastFilter.Update()
    
    # Create actors
    inputActor = vtk.vtkImageActor()
    inputActor.GetMapper().SetInputConnection(inputCastFilter.GetOutputPort())
    
    normalizedActor = vtk.vtkImageActor()
    normalizedActor.GetMapper().SetInputConnection(normalizeCastFilter.GetOutputPort())
    
    # There will be one render window
    renderWindow = vtk.vtkRenderWindow()
    renderWindow.SetSize(600, 300)
    
    # And one interactor
    interactor = vtk.vtkRenderWindowInteractor()
    interactor.SetRenderWindow(renderWindow)
    
    # Define viewport ranges
    # (xmin, ymin, xmax, ymax)
    leftViewport = [0.0, 0.0, 0.5, 1.0]
    rightViewport = [0.5, 0.0, 1.0, 1.0]
    
    # Setup both renderers
    leftRenderer = vtk.vtkRenderer()
    renderWindow.AddRenderer(leftRenderer)
    leftRenderer.SetViewport(leftViewport)
    leftRenderer.SetBackground(colors.GetColor3d("left_bkg"))
    
    rightRenderer = vtk.vtkRenderer()
    renderWindow.AddRenderer(rightRenderer)
    rightRenderer.SetViewport(rightViewport)
    rightRenderer.SetBackground(colors.GetColor3d("right_bkg"))
    
    leftRenderer.AddActor(inputActor)
    rightRenderer.AddActor(normalizedActor)
    
    leftRenderer.ResetCamera()
    
    rightRenderer.ResetCamera()
    
    renderWindow.Render()
    interactor.Start()

if __name__ == '__main__':
    main()
